packages/core/android/src/main/kotlin/com/datadog/reactnative/DdSdkConfigurationExt.kt

```diff
@@ -94,13 +95,27 @@ internal fun ReadableMap.asProxyConfig(): Pair<Proxy, ProxyAuthenticator?>? {
 }
 
 internal fun ReadableArray.asFirstPartyHosts(): Map<String, Set<TracingHeaderType>> {
-    val firstPartyHosts = this.toArrayList() as List<ReadableMap>
-    return firstPartyHosts.mapNotNull<ReadableMap, JSONFirstPartyHost> {
-        val match = it.getString("match")
-        val propagatorTypes = it.getArray("propagatorTypes")
-        if (match != null && propagatorTypes != null) {
-            JSONFirstPartyHost(match, propagatorTypes.toArrayList() as List<String>)
-        } else null
+    return this.toList().mapNotNull {
+        if (it == null) {
+            null
+        } else if (it !is Map<*, *>) {
```

Comment:
`mapNotNull` already guarantees that argument is not `null`

```suggestion
    return this.toList().mapNotNull {
        if (it !is Map<*, *>) {
```

Reply:
`mapNotNull` only guarantees that the resulting list will not contain any null values, but the argument can be null, and when it is I want to skip it, so I return null.

Reply:
indeed, I'm wrong here, ignore my comment then. Alternatively you can call `filterNotNull` btw to avoid having this check.